### PR TITLE
Solves issues with overlays by updating the whole subsystem

### PR DESCRIPTION
**Pull request by SandPoot**
## About The Pull Request
Port of https://github.com/tgstation/tgstation/pull/58378.

## Why It's Good For The Game
Supposedly good, silicons approves.

## Changelog
:cl:
refactor: Refactored the overlay subsystem.
/:cl:

### CODE REVIEW

**Comment by SandPoot** [comment]
A.

**Comment by DeltaFire15** [comment]
Fails CI.

**Comment by SandPoot** [comment]
I need to figure out all those stupid atoms missing some overlay for something.

**Comment by SandPoot** [comment]
@DeltaFire15 unstale and doesn't fail CI :)

File: code/__DEFINES/_flags/_flags.dm [hunk 1]
```diff
@@ -26,32 +26,30 @@ GLOBAL_LIST_INIT(bitflags, list(
...
+#define INITIALIZED_1				(1<<9)
 ///was this spawned by an admin? used for stat tracking stuff.
-#define ADMIN_SPAWNED_1				(1<<15)
+#define ADMIN_SPAWNED_1				(1<<10)
 /// should not get harmed if this gets caught by an explosion?
-#define PREVENT_CONTENTS_EXPLOSION_1 (1<<16)
+#define PREVENT_CONTENTS_EXPLOSION_1 (1<<11)
 /// Early returns mob.face_atom()
-#define BLOCK_FACE_ATOM_1			(1<<17)
+#define BLOCK_FACE_ATOM_1			(1<<12)
```
Comment (deathride58) [inline code/__DEFINES/_flags/_flags.dm]: These are all tied to `flags_1`, an atom var, as are the below turf flags. Wouldn't this cause conflicts where, for instance, anything tagged with `HEAR_1` is also `NO_JAUNT_1`, and anything with `UNUSED_RESERVATION_TURF_1` is also `DEFAULT_RICOCHET_1`?

Comment (SandPoot) [reply code/__DEFINES/_flags/_flags.dm]: Oh, it might conflict? I need to check some stuff.

**Review by deathride58 — APPROVED** [review]
Two whole years of development,,,,

**Pull request merged**